api/nodes/permissions.py

```diff
@@ -22,7 +24,9 @@ def has_object_permission(self, request, view, obj):
 class AdminOrPublic(permissions.BasePermission):
 
     def has_object_permission(self, request, view, obj):
-        assert isinstance(obj, (Node, User, Institution)), 'obj must be a Node, User or Institution, got {}'.format(obj)
+        if isinstance(obj, dict):
```

Comment:
Your helper does this conditional implicitly. Maybe consider removing the `if` here and above.